### DISCUSSION TITLE
What does a "group" relates to?

### DISCUSSION THREAD

**Discussion by wsw70** [jamesoff/simplemonitor#1428] [Q&A]
As I progress in my monitoring, I realized that the HTML dashboard was empty. After some debugging and scratching my head a lot I found out that when I add a default **alerting** group, it was used for **logging**.

Concretely, this is what I had in `monitor.ini`:
```
[reporting]
loggers=db,file,status,json_full
alerters=ntfy.sh
```

In `monitors.ini` I was declaring a default group for **alerting** (at least in my imagination):
```
[defaults]
tolerance=2
group=ntfy.sh
```

The effect? No **logging** (alerting was fine). `save_results2()` was never called.

I got rid of the groups (no groups and no defaults) and everything works.

Thus my question: gow does `group` work? What should I have there? Does it apply to both logging and alerting?

**Comment by jamesoff** [comment]
Groups are ways to put monitors with specific alerters and loggers. Monitors can have a single group, and Loggers and Alerters can have multiple ones. If it's not specified, there's an implicit group called `default` which is used for things so that all Loggers and Alerters work with all monitors.

If you want to have a Logger work for all your Monitors despite some of them having groups, you can specify your Logger as being in the special group `_all` (with the underscore at the front) which matches all monitor groups.
